src/ucs/config/parser.c

```diff
@@ -1131,38 +1152,47 @@ int ucs_config_sscanf_key_value(const char *buf, void *dest, const void *arg)
     while (NULL != token) {
         ucs_string_split(token, VALUE_DELIM, 2, &key, &value);
 
-        /* Remember last value in the values array */
-        if (UCS_OK != ucs_config_key_value_set(param->keys, values, key_count,
-                                               key, value)) {
-            return 0;
+        if (NULL == value) {
+            common_value = key;
+        } else {
+            if (UCS_OK != ucs_config_key_value_set(param, dest, key_set,
+                                                   key, value)) {
+                goto err;
+            }
         }
+
         tokens = ucs_string_split(tokens, TOKEN_DELIM, 1, &token);
     }
 
-    /* Verify that there is a value for each key. Do it before actual set, so
-     * that we don't need to revert in case of failure */
+    /* Set common value if any */
+    if (NULL != common_value) {
+        if (UCS_OK != ucs_config_key_value_set_common(param, dest, key_set,
```

Comment:
the wrapper ucs_config_key_value_set_common seems redundant
after moving its code here we can unite it with next loop:

```
for (index = 0; index < key_count; ++index) {
    if (key_set[index]) {
    } else if (NULL != common_value) {
       parser->read(..., common_value);
    } else {
       error("no value configured...");
    }
}
```

Reply:
ok